static/js/publisher/pages/Builds/__tests__/RepoSelector.test.tsx

```diff
@@ -0,0 +1,406 @@
+      }),
+    );
+
+    renderComponent();
+
+    const user = userEvent.setup();
+    const orgSelect = screen.getByLabelText("Select an organization");
+
+    // Select an organization
+    await user.selectOptions(orgSelect, "canonical");
+
+    // Wait for repos to load and autofill to happen
+    await waitFor(() => {
+      const repoInput = screen.getByLabelText(
+        "Select a repository",
+      ) as HTMLInputElement;
+      expect(repoInput.value).toBe("test-snap-id");
+    });
+  });
+
+  test("does not autofill when no matching repository exists", async () => {
+    const reposWithoutMatch = [
+      { name: "other-repo", nameWithOwner: "johndoe/other-repo" },
+      { name: "another-repo", nameWithOwner: "johndoe/another-repo" },
+    ];
+
+    server.use(
+      http.get("/publisher/github/get-repos", () => {
+        return HttpResponse.json(reposWithoutMatch);
+      }),
+    );
+
+    renderComponent();
+
+    const user = userEvent.setup();
+    const orgSelect = screen.getByLabelText("Select an organization");
+
+    // Select the user's personal organization
+    await user.selectOptions(orgSelect, "johndoe");
+
+    // Wait for repos to load by checking the input is enabled
+    await waitFor(() => {
+      const repoInput = screen.getByLabelText(
+        "Select a repository",
+      ) as HTMLInputElement;
+      expect(repoInput).not.toBeDisabled();
+    });
+
+    // Repository input should remain empty
+    const repoInput = screen.getByLabelText(
+      "Select a repository",
+    ) as HTMLInputElement;
+    expect(repoInput.value).toBe("");
+  });
+
+  test("clears input when organization changes", async () => {
+    server.use(
+      http.get("/publisher/github/get-repos", ({ request }) => {
+        const url = new URL(request.url);
+        const org = url.searchParams.get("org");
+        if (org === "canonical") {
+          return HttpResponse.json(mockOrgRepos);
+        }
+        return HttpResponse.json(mockRepos);
+      }),
+      http.get("/api/test-snap-id/builds/validate-repo", () => {
+        return HttpResponse.json({ success: true });
+      }),
+    );
+
+    renderComponent();
+
+    const user = userEvent.setup();
+    const orgSelect = screen.getByLabelText("Select an organization");
+
+    // Select user organization first
+    await user.selectOptions(orgSelect, "johndoe");
+
+    // Wait for autofill
+    await waitFor(() => {
+      const repoInput = screen.getByLabelText(
+        "Select a repository",
+      ) as HTMLInputElement;
+      expect(repoInput.value).toBe("test-snap-id");
+    });
+
+    // Change to different organization
+    await user.selectOptions(orgSelect, "canonical");
+
+    // Input should be cleared and then autofilled again
+    await waitFor(() => {
+      const repoInput = screen.getByLabelText(
+        "Select a repository",
+      ) as HTMLInputElement;
+      expect(repoInput.value).toBe("test-snap-id");
```

Comment:
How do we know it was cleared and filled again? If the value is exactly the same as in previous `expect` it could have not changed at all.